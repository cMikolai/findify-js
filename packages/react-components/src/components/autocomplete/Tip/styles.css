.tip {
  user-select: none;
  cursor: pointer;
  text-align: center;
  height: 48px;
  line-height: 48px;
  font-family: $font-base;
  font-size: $font-size-large;
  color: #b2b2b2;
  background: $color-grey-1;
}

.highlight {
  color: #404040;
}
