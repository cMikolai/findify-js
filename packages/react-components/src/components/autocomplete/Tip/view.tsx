import React from 'react'
import classnames from 'classnames'

import styles from './styles.css'

export default ({ query, className, title }) => (
  <div className={classnames(styles.root, className)}>
    {title}
    {' '}
    {
      query.length > 0
      ? <span className={styles.highlight}>{query}</span>
      : ''
    }
  </div>
)
