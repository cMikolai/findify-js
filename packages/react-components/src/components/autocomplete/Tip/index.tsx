import view from './view'
import { connectQuery } from '@findify/react-connect';

export default connectQuery(view)
