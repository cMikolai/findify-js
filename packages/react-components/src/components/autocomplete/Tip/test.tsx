import React from 'react'
import Tip from './view'
import { shallow } from 'enzyme'

const config = {
  get: () => 'Press Enter to search'
}

describe('Tip', () => {
  it('renders correctly', () => {
    expect(
      shallow(
        <Tip className='test' query='' title='Press Enter to search' />
      )
    ).toMatchSnapshot()

    expect(
      shallow(
        <Tip className='test' query='qu' title='Press Enter to search' />
      )
    ).toMatchSnapshot()
  })
})
