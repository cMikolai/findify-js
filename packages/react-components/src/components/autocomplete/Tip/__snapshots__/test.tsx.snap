// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tip renders correctly 1`] = `
<div
  className="root test"
>
  Press Enter to search
   
</div>
`;

exports[`Tip renders correctly 2`] = `
<div
  className="root test"
>
  Press Enter to search
   
  <span
    className="highlight"
  >
    qu
  </span>
</div>
`;
